lexer/lexer.go

```diff
@@ -36,66 +37,136 @@ func New(source string) *Lexer {
 	}
 }
 
-func (l *Lexer) NextChar() byte {
-	// TODO: Check overflow using sourceLength here
-	l.currentChar = l.Source[l.index]
+func (l *Lexer) willOverflow() bool {
+	return l.index+1 >= l.sourceLength
+}
+
+// NOTE: This might need to rune, depending on what character set jsonet supports.
+func (l *Lexer) CurrentChar() byte {
+	return l.Source[l.index]
+}
+
+func (l *Lexer) NextChar() (byte, error) {
+	if l.willOverflow() {
+		// TODO: Zero valued bytes, how?
+		return '\x00', errors.New("Lexer will overflow")
```

Comment:
the only other way I can think of doing it would be to declare char above with `var char byte` and then return `char` everywhere.